src/utils/portfolioCalculations/calculateRealizedProfit.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  calculationParams,
+  CurrentPortfolioTransaction,
+} from '@/types/calculations';
+import {
+  getSortedPurchaseTransactions,
+  getSortedSaleTransactions,
+} from './getSortedTransactions';
+import { FX_RATE } from './const/FX_RATE';
+
+export const calculateRealizedProfit = ({
+  existingTransactions,
+  savedTransactions,
+}: calculationParams) => {
+  const transactions = [...existingTransactions, ...savedTransactions];
+
+  const purchaseTransactions: CurrentPortfolioTransaction[] =
+    getSortedPurchaseTransactions(transactions).map((tx) => ({
+      ...tx,
+      remainingNumberOfStocksOwned: tx.numberOfStocks,
+    }));
+  const saleTransactions = getSortedSaleTransactions(transactions).map(
+    (tx) => ({
+      ...tx,
+      remainingNumberOfStocksToSell: tx.numberOfStocks,
+    }),
+  );
+
+  let realizedProfit = 0;
+
+  saleTransactions.forEach((saleTx) => {
+    let remainingNumberOfStocksToSell = saleTx.remainingNumberOfStocksToSell;
+
+    purchaseTransactions.forEach((purchaseTx) => {
```

Comment:
I think both of these `forEach` can be replaced with `reduce`. That would eliminate the mutable parts of the code.